File: Python/bubbleSort.py
class BubbleSort:
    def __init__(self, arr: list) -> None:
        self.arr = arr

    def sorting(self) -> list:
        for i in range(len(self.arr) - 1):
            for j in range(len(self.arr) - i - 1):
                if self.arr[j] > self.arr[j + 1]:
                    temp = self.arr[j]
                    self.arr[j] = self.arr[j + 1]
                    self.arr[j + 1] = temp

        return self.arr


print(BubbleSort([24, 23, 34, 2, 23, 53, 342]).sorting())

File: Go/BubbleSort.go
package main

import "fmt"

// BubbleSort function that sorts an integer slice in ascending order
func BubbleSort(arr []int) {
    n := len(arr)
    for i := 0; i < n-1; i++ {
        // Flag to check if any swap happens
        swapped := false
        for j := 0; j < n-i-1; j++ {
            if arr[j] > arr[j+1] {
                // Swap arr[j] and arr[j+1]
                arr[j], arr[j+1] = arr[j+1], arr[j]
                swapped = true
            }
        }
        // If no elements were swapped, the array is already sorted
        if !swapped {
            break
        }
    }
}

func main() {
    arr := []int{64, 34, 25, 12, 22, 11, 90}
    fmt.Println("Unsorted array:", arr)
    BubbleSort(arr)
    fmt.Println("Sorted array:", arr)
}
